contracts/src/chainlink/solidity/emergency/StarkNetValidator.sol

```diff
@@ -63,6 +73,14 @@ contract StarkNetValidator is TypeAndVersionInterface, AggregatorValidatorInterf
     return "StarkNetValidator 0.1.0";
   }
 
+  function setFee(uint256 newFee) external onlyOwner {
```

Comment:
https://docs.starknet.io/docs/L1-L2%20Communication/messaging-mechanism#l1--l2-message-fees

>  The sequencer charges the fee in full upon updating the L1 state with the consumption of this message.

Static fees will not work for us I'm afraid as it looks like the fee will be taken in full. On Arbitrum the static fee is an option and it currently works that way, as we can overpay for the message send and get the excess refunded.

Reply:
Ok, here is a strategy I think we should try.

Let's try to express the bridge fee as a function of the L1 gas price: https://data.chain.link/ethereum/mainnet/gas/fast-gas-gwei

We should be able to set and configure:

- L1 gas price feed address
- L2 tx gas estimate, which we can figure out empirically via https://starknet.io/docs/hello_starknet/l1l2.html#estimate-message-fee

When configuring the L2 tx gas estimate, we can add some buffer on top to be able to mitigate more volatility.

Additionally, I think we should add the **ability for the owner to retry** the last transmit/validate/send, if for whatever reason the L1 -> L2 message doesn't go through. To implement this I think we should store the L1 feed that we validate, and read the last round from there on retry (onlyOwner).